Core/include/Acts/Propagator/TryAllOverstepNavigator.hpp

```diff
@@ -0,0 +1,529 @@
+    std::size_t activeCandidateIndex = 0;
+
+    /// The position before the last step
+    std::optional<Vector3> lastPosition;
+    /// The last intersection used to avoid rehitting the same surface
+    std::optional<SurfaceIntersection> lastIntersection;
+
+    /// Indicator if the target is reached
+    bool targetReached = false;
+    /// If a break has been detected
+    bool navigationBreak = false;
+
+    /// Provides easy access to the active intersection candidate
+    const detail::IntersectionCandidate& activeCandidate() const {
+      return activeCandidates.at(activeCandidateIndex);
+    }
+  };
+
+  /// Constructor with configuration object
+  ///
+  /// @param cfg The navigator configuration
+  /// @param _logger a logger instance
+  TryAllOverstepNavigator(Config cfg,
+                          std::unique_ptr<const Logger> _logger =
+                              getDefaultLogger("TryAllOverstepNavigator",
+                                               Logging::INFO))
+      : m_cfg(std::move(cfg)), m_logger{std::move(_logger)} {}
+
+  State makeState(const Surface* startSurface,
+                  const Surface* targetSurface) const {
+    State result;
+    result.startSurface = startSurface;
+    result.targetSurface = targetSurface;
+    return result;
+  }
+
+  const Surface* currentSurface(const State& state) const {
+    return state.currentSurface;
+  }
+
+  const TrackingVolume* currentVolume(const State& state) const {
+    return state.currentVolume;
+  }
+
+  const IVolumeMaterial* currentVolumeMaterial(const State& state) const {
+    if (state.currentVolume == nullptr) {
+      return nullptr;
+    }
+    return state.currentVolume->volumeMaterial();
+  }
+
+  const Surface* startSurface(const State& state) const {
+    return state.startSurface;
+  }
+
+  const Surface* targetSurface(const State& state) const {
+    return state.targetSurface;
+  }
+
+  bool targetReached(const State& state) const { return state.targetReached; }
+
+  bool endOfWorldReached(State& state) const {
+    return state.currentVolume == nullptr;
+  }
+
+  bool navigationBreak(const State& state) const {
+    return state.navigationBreak;
+  }
+
+  void currentSurface(State& state, const Surface* surface) const {
+    state.currentSurface = surface;
+  }
+
+  void targetReached(State& state, bool targetReached) const {
+    state.targetReached = targetReached;
+  }
+
+  void navigationBreak(State& state, bool navigationBreak) const {
+    state.navigationBreak = navigationBreak;
+  }
+
+  /// @brief Initialize call - start of navigation
+  ///
+  /// @tparam propagator_state_t The state type of the propagator
+  /// @tparam stepper_t The type of stepper used for the propagation
+  ///
+  /// @param [in,out] state is the propagation state object
+  /// @param [in] stepper Stepper in use
+  template <typename propagator_state_t, typename stepper_t>
+  void initialize(propagator_state_t& state, const stepper_t& stepper) const {
```

Comment:
This looks very similar to the `initialize` function of the `TryAllNavigator` (except the last two lines). Maybe this can be unified as a function?

Another option would be to share code between the two try-all-navigator` via (static) inheritance, since other small functions are duplicated as well...